### PR TITLE
Use promises and ES2015ify tests

### PR DESCRIPTION
**Pull request by kevva**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,33 +11,29 @@ function unobfuscateEmail(str) {
 		.join('');
 }
 
-module.exports = function (username, cb) {
+module.exports = function (username) {
 	if (typeof username !== 'string') {
 		throw new Error('username required');
```
Comment (SamVerschueren) [inline index.js]: Same as the others. Return rejected promise here.

File: package.json [hunk 2]
```diff
@@ -37,10 +37,16 @@
...
+    "got": "^4.2.0",
     "meow": "^3.3.0"
   },
   "devDependencies": {
-    "ava": "0.0.4"
+    "ava": "*",
+    "xo": "*"
+  },
+  "xo": {
+    "ignore": [
```
Comment (SamVerschueren) [inline package.json]: Should be [`ignores`](https://github.com/sindresorhus/xo#ignores) according to the docs. Out of my own experience, `ignore` works as well but I think we should be consistent in this one :).

File: index.js (hunk 1 above)
Comment (kevva) [inline index.js]: Whoops.

File: index.js [hunk 3]
```diff
@@ -11,33 +12,29 @@ function unobfuscateEmail(str) {
...
 			name: $('.fullname').text() || null,
 			email: unobfuscateEmail($('.email [data-email]').attr('data-email')) || null,
 			homepage: $('.homepage a').attr('href') || null,
 			github: $('.github a').text().slice(1) || null,
 			twitter: $('.twitter a').text().slice(1) || null,
 			freenode: $('.freenode a').text() || null
-		});
+		};
+	}).catch(function (err) {
+		if (err.code === 404) {
```
Comment (sindresorhus) [inline index.js]: `err.code` => `err.statusCode`

**Comment by sindresorhus** [comment]
Can you fix the merge conflict? Sorry, I pushed an update forgetting about this PR...

**Comment by kevva** [comment]
Should be resolved now.

File: test.js [hunk 4]
```diff
@@ -1,18 +1,14 @@
...
-		console.log(user);
-		t.assert(!err, err);
-		t.assert(user.name === 'Sindre Sorhus');
-		t.assert(user.email === 'sindresorhus@gmail.com');
-		t.assert(user.homepage === 'http://sindresorhus.com');
-		t.assert(user.github === 'sindresorhus');
-		t.assert(user.twitter === 'sindresorhus');
-		t.assert(user.freenode === 'sindresorhus');
-	});
+	console.log(user);
```
Comment (SamVerschueren) [inline test.js:7]: Any reason why the user is logged here? Or is this something that was added just to see what properties the object has that could be tested?

Just wondering :).

Comment (kevva) [inline test.js:7]: @sindresorhus could probably answer for that.

Comment (sindresorhus) [inline test.js:7]: Previously, the assert methods didn't show what output if it didn't match, so just logged it as a temp workaround.

Comment (SamVerschueren) [inline test.js:7]: So it can be removed now?

Comment (sindresorhus) [inline test.js:7]: Yup, gonna do it.

Comment (SamVerschueren) [inline test.js:7]: :+1:

**Pull request merged**